src/control/system/database_pools.go

```diff
@@ -43,13 +46,25 @@ type (
 	// PoolServiceState is used to represent the state of the pool service
 	PoolServiceState uint
 
+	// PoolServiceStorage holds information about the pool storage.
+	PoolServiceStorage struct {
+		sync.Mutex
+		CreationRankStr string   // string rankset set at creation
+		creationRanks   *RankSet // used to reconstitute the rankset
+		CurrentRankStr  string   // string rankset representing current ranks
+		currentRanks    *RankSet // used to reconstitute the rankset
+		ScmPerRank      uint64   // scm per rank allocated during creation
+		NVMePerRank     uint64   // nvme per rank allocated during creation
+	}
+
 	// PoolService represents a pool service created to manage metadata
 	// for a DAOS Pool.
 	PoolService struct {
```

Comment:
Raft term field was requested in DAOS-6088, presumably that should be added here as well so that pool service replica ranks update events can provide raft term value.

Reply:
Hmm, yes, I think you're right about that. We want it to be sent as part of the message so that we can ensure that we're only making an update if the term is greater than the current value. That having been said, I'd prefer to add that in another PR that is focused on handling the service rank events.